manager/service.go

```diff
@@ -239,7 +241,7 @@ func (ms *managerService) CreateVM(ctx context.Context, req *CreateReq) (string,
 				if err := ms.RemoveVM(ctx, id); err != nil {
 					ms.logger.Error("Failed to remove VM after TTL", "error", err)
 				}
-			case <-ctx.Done():
+			case <-ms.ctx.Done():
```

Comment:
Context is usually not part of the structure, but a param to a function. I.e. context provides an execution context, usually not the whole "world" context. It indicates design problems. Let's merge, but this should be considered in refactoring tasks, so I'll create a ticket out of this comment.

Reply:
We cannot use context from the function here since it's the grpc request context and will be cancelled when the function returns. But we still need context since we spin up this go routine and don't want to leave it dangling if manager service shuts down before the goroutine returns